### PR TITLE
[12.0][ADD] sale_timesheet_hook: add hook points in sale_timesheets

### PR DESCRIPTION
**Pull request by alexey-pelykh**
This module adds hook points in `sale_timesheets` module in order to add more
flexibility in the billing types for the creation of *Sale Orders*.

Although the original code has been modified a bit, the logic has been
respected and the method still does exactly the same.

Technical Enhancements:
 * `account.analytic.line` extendability improvements:
   * `_get_valid_so_line_ids` overridable
 * `project.create.sale.order` extendability improvements:
   * `action_create_sale_order` refactor, reported in odoo/odoo#31047.

### CODE REVIEW

**Comment by alexey-pelykh** [comment]
Supersedes https://github.com/OCA/sale-workflow/pull/762 @pedrobaeza @victormmtorres @rafaelbn

File: sale_timesheet_hook/README.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_timesheet_hook
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/167/12.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module adds hook points in ``sale_timesheets`` module in order to add more
```
Comment (pedrobaeza) [inline sale_timesheet_hook/README.rst]: Module is `sale_timesheet`

File: sale_timesheet_hook/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    'category': 'Sales',
+    'website': 'https://github.com/OCA/sale-workflow',
+    'author':
+        'Brainbean Apps, '
+        'Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'installable': True,
+    'application': False,
+    'summary': (
+        'Adds hook points in sale_timesheets module in order to add more'
```
Comment (pedrobaeza) [inline sale_timesheet_hook/__manifest__.py]: sale_timesheet

File: sale_timesheet_hook/README.rst (hunk 1 above)
Comment (alexey-pelykh) [reply sale_timesheet_hook/README.rst]: Fixed

File: sale_timesheet_hook/__manifest__.py (hunk 2 above)
Comment (alexey-pelykh) [reply sale_timesheet_hook/__manifest__.py]: Fixed

**Review by pedrobaeza — APPROVED** [review]

**Review by nikul-serpentcs — APPROVED** [review]
Only Code Review 
LGTM :+1:

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: LGTM

**Pull request merged**